airflow/models/dagrun.py

```diff
@@ -640,15 +640,9 @@ def task_instance_scheduling_decisions(self, session: Session = NEW_SESSION) ->
         tis = list(self.get_task_instances(session=session, state=State.task_states))
         self.log.debug("number of tis tasks for %s: %s task(s)", self, len(tis))
         dag = self.get_dag()
-        for ti in tis:
-            try:
-                ti.task = dag.get_task(ti.task_id)
-            except TaskNotFound:
-                self.log.warning(
-                    "Failed to get task '%s' for dag '%s'. Marking it as removed.", ti, ti.dag_id
-                )
-                ti.state = State.REMOVED
-                session.flush()
+
+        if self.should_verify_integrity(dag, tis, session=session):
```

Comment:
Should this check be a part of `verify_integrity`?

Reply:
I think it should not be part of it. Running the verify integrity only when there're missing indexes would save time...